backend/src/routes/api/notebooks/index.ts

```diff
@@ -49,6 +51,19 @@ module.exports = async (fastify: KubeFastifyInstance) => {
           ? false
           : await getNotebookStatus(fastify, namespace, name);
 
+        const notebookName = notebook.metadata.name;
+        if (isRunning && !notebook.metadata.annotations['opendatahub.io/link']) {
+          getRoute(fastify, namespace, notebookName)
+            .then((route) => {
+              patchNotebookRoute(fastify, route, namespace, notebookName).catch((e) => {
+                fastify.log.warn(`Failed patching route to notebook ${notebookName}, ${e.message}`);
+              });
+            })
+            .catch((e) => {
+              fastify.log.warn(`Failed getting route ${notebookName}, ${e.message}`);
```

Comment:
Your Route logic doesn't handle the API error... but `patchNotebookRoute` does and converts it to an object.

Best to add a `.catch` on the Route call in `getRoute` and convert it over to a `createCustomError` object like we do in the patch call.

---

For Kube calls we make on the backend, the actual message is hidden under `e.response.body.status` which is the status object. `e.message` on the kube response error is just going to be `HTTP request failed`